cmd/test/backend_kv_test.cpp

```diff
@@ -0,0 +1,279 @@
+    return out;
+}
+
+UnaryStats unary_stats;
+
+class AsyncCall {
+  public:
+    explicit AsyncCall(std::shared_ptr<grpc::Channel> channel, grpc::CompletionQueue* queue)
+    : channel_(channel), queue_(queue) {}
+    virtual ~AsyncCall() = default;
+
+    virtual void handle_completion(bool ok) = 0;
+
+    std::chrono::steady_clock::time_point start_time() const { return start_time_; }
+
+  protected:
+    grpc::ClientContext client_context_;
+    std::shared_ptr<grpc::Channel> channel_;
+    grpc::CompletionQueue* queue_;
+    std::chrono::steady_clock::time_point start_time_;
+};
+
+template <typename Reply>
+using AsyncResponseReaderPtr = std::unique_ptr<grpc::ClientAsyncResponseReaderInterface<Reply>>;
+
+template<
+    typename Request,
+    typename Reply,
+    AsyncResponseReaderPtr<Reply>(remote::ETHBACKEND::StubInterface::*PrepareAsync)(grpc::ClientContext*, const Request&, grpc::CompletionQueue*)
+>
+class AsyncUnaryCall : public AsyncCall {
+  public:
+    explicit AsyncUnaryCall(std::shared_ptr<grpc::Channel> channel, grpc::CompletionQueue* queue)
+    : AsyncCall(channel, queue), stub_(remote::ETHBACKEND::NewStub(channel)) {}
+
+    void start_async() {
+        SILK_DEBUG << "AsyncUnaryCall::start_async START";
+        auto response_reader_ = (stub_.get()->*PrepareAsync)(&client_context_, Request{}, queue_);
+        response_reader_->StartCall();
+        response_reader_->Finish(&reply_, &status_, this);
+        start_time_ = std::chrono::steady_clock::now();
+        ++unary_stats.started_count;
+        SILK_TRACE << "AsyncUnaryCall::start_async END";
+    }
+
+  protected:
+    std::unique_ptr<remote::ETHBACKEND::StubInterface> stub_;
+    grpc::Status status_;
+    Reply reply_;
+};
+
+class AsyncEtherbaseCall : public AsyncUnaryCall<
+        remote::EtherbaseRequest, remote::EtherbaseReply, &remote::ETHBACKEND::StubInterface::PrepareAsyncEtherbase> {
+  public:
+    explicit AsyncEtherbaseCall(std::shared_ptr<grpc::Channel> channel, grpc::CompletionQueue* queue)
+    : AsyncUnaryCall(channel, queue) {}
+
+    void handle_completion(bool ok) override {
+        SILK_DEBUG << "AsyncEtherbaseCall::handle_completion ok: " << ok << " status: " << status_;
+        ++unary_stats.completed_count;
+        if (ok && status_ == grpc::Status::OK) {
+            if (reply_.has_address()) {
+                const auto h160_address = reply_.address();
+                SILK_INFO << "Etherbase reply: " << silkworm::to_hex(address_from_H160(h160_address));
+            } else {
+                SILK_INFO << "Etherbase reply: no address";
+            }
+            ++unary_stats.ok_count;
+        } else {
+            ++unary_stats.ko_count;
+        }
+    }
+};
+
+void start_unary_async(std::atomic_bool& stop, std::shared_ptr<grpc::Channel> channel, grpc::CompletionQueue* queue, int batch_size) {
+    for (auto i{0}; i<batch_size && !stop; i++) {
+        SILK_DEBUG << "New Etherbase async call starting...";
```

Comment:
Just asking ... isn't it better to use trace for logging function entrance/exit ?

Reply:
Thanks for pointing this out, made me realize that just one DEBUG log in needed here (TRACE logs are within `start_async` method).